crates/router/src/connector/adyen.rs

```diff
@@ -801,3 +802,103 @@ impl api::IncomingWebhook for Adyen {
         })
     }
 }
+
+impl api::Validator<api::Global> for Adyen {
+    fn validate_metadata(
+        &self,
+        payment_method: Option<(api::enums::PaymentMethod, api::enums::PaymentMethodType)>,
+        metadata: serde_json::Value,
+    ) -> CustomResult<(), common_utils::errors::ValidationError> {
+        if let Some((api::enums::PaymentMethod::Wallet, api::enums::PaymentMethodType::GooglePay)) =
+            payment_method
+        {
+            let _inner: api_models::payments::GpaySessionTokenData = metadata
+                .parse_value("GpayTokenParameters")
+                .change_context(errors::ValidationError::InvalidValue {
+                    message: "Failed while getting google pay metadata".to_string(),
+                })?;
+        }
+
+        Ok(())
```

Comment:
Seems like this implementation is same in most the places so can it be the default impl in the trait ?

Reply:
True, but we cannot have this as default as the validation that is done here is specific to a connector. This was decided based on the features that are implemented or possible for a certain connector